Build fails with gcc > 6 (will PR the fix from the Debian team)

I'm building eXpress within the [Spack](https://spack.io) framework.

Building express fails with the following:

```
     383    /home/ghartzell/spack/var/spack/stage/express-1.5.2-6423bq5bjovjwpq2hc4tyi6lwslzubv7/eXpress-1.5.2/src/logger.h: In member function 'void Logger::severe
            (const char*, ...) const':
     384    /home/ghartzell/spack/var/spack/stage/express-1.5.2-6423bq5bjovjwpq2hc4tyi6lwslzubv7/eXpress-1.5.2/src/logger.h:72:37: warning: unnecessary parentheses
            in declaration of '_mut' [-Wparentheses]
     385         boost::unique_lock<boost::mutex>(_mut);
     386                                         ^
     387    /home/ghartzell/spack/var/spack/stage/express-1.5.2-6423bq5bjovjwpq2hc4tyi6lwslzubv7/eXpress-1.5.2/src/targets.cpp: In member function 'double Target::s
            ample_likelihood(bool, const std::vector<const Target*>*) const':
  >> 388    /home/ghartzell/spack/var/spack/stage/express-1.5.2-6423bq5bjovjwpq2hc4tyi6lwslzubv7/eXpress-1.5.2/src/targets.cpp:116:62: error: no matching function f
            or call to 'Target::cached_effective_length(const boost::shared_ptr<BiasBoss>&) const'
     389       double tot_eff_len = cached_effective_length(lib.bias_table);
     390                                                                  ^
     391    In file included from /home/ghartzell/spack/var/spack/stage/express-1.5.2-6423bq5bjovjwpq2hc4tyi6lwslzubv7/eXpress-1.5.2/src/targets.cpp:10:
     392    /home/ghartzell/spack/var/spack/stage/express-1.5.2-6423bq5bjovjwpq2hc4tyi6lwslzubv7/eXpress-1.5.2/src/targets.h:399:10: note: candidate: 'double Target
            ::cached_effective_length(bool) const'
     393       double cached_effective_length(bool with_bias=true) const;
     394              ^~~~~~~~~~~~~~~~~~~~~~~
     395    /home/ghartzell/spack/var/spack/stage/express-1.5.2-6423bq5bjovjwpq2hc4tyi6lwslzubv7/eXpress-1.5.2/src/targets.h:399:10: note:   no known conversion for
             argument 1 from 'const boost::shared_ptr<BiasBoss>' to 'bool'
  >> 396    /home/ghartzell/spack/var/spack/stage/express-1.5.2-6423bq5bjovjwpq2hc4tyi6lwslzubv7/eXpress-1.5.2/src/targets.cpp:121:77: error: no matching function f
            or call to 'Target::cached_effective_length(const boost::shared_ptr<BiasBoss>&) const'
     397                                 neighbor->cached_effective_length(lib.bias_table));
     398                                                                                 ^
     399    In file included from /home/ghartzell/spack/var/spack/stage/express-1.5.2-6423bq5bjovjwpq2hc4tyi6lwslzubv7/eXpress-1.5.2/src/targets.cpp:10:
     400    /home/ghartzell/spack/var/spack/stage/express-1.5.2-6423bq5bjovjwpq2hc4tyi6lwslzubv7/eXpress-1.5.2/src/targets.h:399:10: note: candidate: 'double Target
            ::cached_effective_length(bool) const'
     401       double cached_effective_length(bool with_bias=true) const;
     402              ^~~~~~~~~~~~~~~~~~~~~~~

```

There's  a fix for this problem in the Debian package repository.  It works well when applied here.  I'll PR it.

--- a/src/targets.cpp
+++ b/src/targets.cpp
@@ -113,12 +113,12 @@ double Target::sample_likelihood(bool with_pseudo,
 
   double ll = LOG_1;
   double tot_mass = mass(with_pseudo);
-  double tot_eff_len = cached_effective_length(lib.bias_table);
+  double tot_eff_len = cached_effective_length(static_cast<bool>(lib.bias_table));
   if (neighbors) {
     foreach (const Target* neighbor, *neighbors) {
       tot_mass = log_add(tot_mass, neighbor->mass(with_pseudo));
       tot_eff_len = log_add(tot_eff_len,
-                            neighbor->cached_effective_length(lib.bias_table));
+                            neighbor->cached_effective_length(static_cast<bool>(lib.bias_table)));
     }
   }
   ll += tot_mass - tot_eff_len;

